### PR TITLE
Add Data Preparation for Report Generation

### PR DESCRIPTION
**Pull request by dongyang0122**
Add detailed data preparation for report generation.

### CODE REVIEW

File: m3/data_prepare/report/2-1_collect-sentence-pool.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+# Configure OpenAI API client
+openai.api_key = os.getenv("OPENAI_API_KEY")  # Ensure the OpenAI API key is set as an environment variable
+if not openai.api_key:
+    raise ValueError("The OpenAI API key is not set. Please set it as an environment variable 'OPENAI_API_KEY'.")
+
+# Constants
+MODEL_NAME = "meta/llama-3.1-8b-instruct"  # or "meta/llama-3.1-70b-instruct"
```
Comment (mingxin-zheng) [inline m3/data_prepare/report/2-1_collect-sentence-pool.py:27]: Would be nice to put the link
https://build.nvidia.com/meta/llama-3_1-8b-instruct
To let developers know where does this string come from

File: m3/data_prepare/report/README.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+Before performing the text conversion, the LLM will be used to analyze the dataset and collect a sample pool of sentences. This pool will consist of commonly occurring phrases or structures found in the medical reports. These standardized phrases guide the text transformation process to ensure consistent input for report generation.
+
+```bash
+python 2-1_collect-sentence-pool.py
+```
+
+#### 2.2 Use LLM and Sentence Pool to Convert Text
+
+In this step, the LLM uses the previously collected sample sentence pool to convert the original medical report text into a standardized format. The LLM will replace or reformat certain sentences and medical terminology to ensure the text is uniform and ready for training the VLM model to generate reliable and coherent medical reports.
```
Comment (mingxin-zheng) [inline m3/data_prepare/report/README.md:43]: Please describe the current implementation uses `openai` to call the LLMs. Developers may use their own locally-deployed ones.

Also, I think you will need to define the `base_url` somewhere, but I couldn't find it. Is the following necessary to call the API?

```
client = OpenAI(
  base_url = "https://integrate.api.nvidia.com/v1",
  api_key = ...
)
```

**Review by holgerroth — COMMENTED** [review]
Nice README. Please address Mingxin's comments @dongyang0122 .

File: m3/data_prepare/report/2-1_collect-sentence-pool.py (hunk 1 above)
Comment (dongyang0122) [reply m3/data_prepare/report/2-1_collect-sentence-pool.py:24]: Thanks. I have added the related information in the scripts.

File: m3/data_prepare/report/README.md (hunk 2 above)
Comment (dongyang0122) [reply m3/data_prepare/report/README.md:46]: Thanks. I have updated the scripts.

**Review by holgerroth — APPROVED** [review]
LGTM

File: m3/data_prepare/report/2-1_collect-sentence-pool.py (hunk 1 above)
Comment (mingxin-zheng) [reply m3/data_prepare/report/2-1_collect-sentence-pool.py:24]: Thanks!

**Pull request merged**